src/test/java/com/olgaboiar/mint/ReaderTest.java

```diff
@@ -0,0 +1,56 @@
+package com.olgaboiar.mint;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ReaderTest {
+
+    @Test
+    void testReaderGetsInputHeadersForRequestWithHeadersOnly() throws IOException {
+        List<String> expected = new ArrayList<String>();
+        String requestLine = "GET /simple_get HTTP/1.1";
+        expected.add(requestLine);
+        StringReader stringReader = new StringReader(requestLine);
+        BufferedReader testBufferReader = new BufferedReader(stringReader);
+        Reader testReader = new Reader(testBufferReader);
+        List<String> testClientInput = testReader.readClientInputHeaders();
+
+        assertEquals(expected, testClientInput);
+    }
+
+    @Test
+    void testReaderGetsInputHeadersForRequestWithHeadersAndBody() throws IOException {
+        List<String> expected = new ArrayList<String>();
+        String requestLine = "GET /simple_get HTTP/1.1\n\nsome_body";
+        String requestHeader = "GET /simple_get HTTP/1.1";
+        expected.add(requestHeader);
+        StringReader stringReader = new StringReader(requestLine);
+        BufferedReader testBufferReader = new BufferedReader(stringReader);
+        Reader testReader = new Reader(testBufferReader);
+        List<String> testClientInput = testReader.readClientInputHeaders();
+
+        assertEquals(expected, testClientInput);
+    }
+
+    @Test
+    void testReaderGetsInputBodyForRequestWithHeadersAndBody() throws IOException {
+        String expected = "some_body";
+        String requestLine = "GET /simple_get HTTP/1.1\nContent-Length: 9\n\nsome_body";
+        StringReader stringReader = new StringReader(requestLine);
+        BufferedReader testBufferReader = new BufferedReader(stringReader);
+        Reader testReader = new Reader(testBufferReader);
+        List<String> testClientInput = testReader.readClientInputHeaders();
+        RequestParser requestParser = new RequestParser();
+        Map<String, String> requestHeaders = requestParser.parseRequestHeaders(testClientInput);
+        int contentLength = Integer.valueOf(requestParser.parseContentLength(requestHeaders));
+        String testClientInputBody = testReader.readClientInputBody(contentLength);
+
+        assertEquals(expected, testClientInputBody);
+    }
+}
```

Comment:
Are there ways we can make these tests smaller? This is a good place to use Mocking.

Reply:
Perhaps we can make the test names more explicit about what we're testing? Think what are the inputs and what are the outputs of the methods you're testing?

Reply:
refactored in https://github.com/olgaboiar/MintHTTPServer/pull/10/commits/00fabc9ee6e69dc1a7c617d39f59b7467593b0ad